### DISCUSSION TITLE
why does bpftool load failed when using bpf_trace_printk in kernel 4.18?

### DISCUSSION THREAD

**Discussion by bianmingkun** [cilium/ebpf#753] [Q&A] [answered]
Hi,

    I want to use bpf_trace_printk, but bpftool will load failed.
    Test code is:
```
#include <linux/bpf.h>
#include <linux/ptrace.h>
#include <linux/version.h>
#include <bpf/bpf_helpers.h>
SEC("kprobe/do_sys_open")
int kprobe__do_sys_open(struct pt_regs *ctx)
{
        char file_name[256];
        char fmt[] = "open file %d\n";
        bpf_trace_printk(fmt, sizeof(fmt), 1);
        return 0;
}
char _license[] SEC("license") = "GPL";

```
Then exec ：
clang -c test1.c -target bpf
./bpftool prog load ./test.o /sys/fs/bpf/test

And error will occur:
libbpf: Program 'kprobe/do_sys_open' contains non-map related relo data pointing to section 6
Error: failed to open object file

It will load successfully when I exec "syscall(__NR_bpf, BPF_PROG_LOAD, attr, size)" by using C code, and bpf_trace_printk  works ok by 
"cat /sys/kernel/debug/tracing/trace".
And bpftool will load successfully too when I delete the code "bpf_trace_printk(fmt, sizeof(fmt), 1)", 

can you give me some suggestions？

kernel version: kernel-4.18.0-147
bpftool: v4.18.0
clang version 10.0.1

Thanks.

**Comment by lmb** [comment, marked as answer]
Hi, if you need support for bpftool emailing bpf@vger.kernel.org is your best bet.
